python/triton/compiler/compiler.py

```diff
@@ -430,6 +430,7 @@ def __init__(self, src, metadata_group, hash):
             file.suffix[1:]: file.read_bytes() if file.suffix[1:] == binary_ext else file.read_text()
             for file in asm_files
         })
+        self.metadata_group = metadata_group
```

Comment:
is this needed?

Reply:
and we need to store it in self?

Reply:
Oh I got what you meant. See the comment a few lines below.

```
 # binaries are lazily initialized
```
Unfortunately we don't know the function pointer when we construct a `CompiledKernel` object. That's why I need to store the `metadata_group` object in `self`